spek-ide-plugin/interop-jvm/src/main/kotlin/org/spekframework/ide/ServiceMessageAdapter.kt

```diff
@@ -0,0 +1,113 @@
+package org.spekframework.ide
+
+import org.spekframework.spek2.runtime.execution.ExecutionListener
+import org.spekframework.spek2.runtime.execution.ExecutionResult
+import org.spekframework.spek2.runtime.scope.ActionScopeImpl
+import org.spekframework.spek2.runtime.scope.GroupScopeImpl
+import org.spekframework.spek2.runtime.scope.TestScopeImpl
+import java.io.CharArrayWriter
+import java.io.PrintWriter
+
+class ServiceMessageAdapter: ExecutionListener {
+    override fun executionStart() { }
+
+    override fun executionFinish() { }
+
+    override fun testExecutionStart(test: TestScopeImpl) {
+        out("testStarted name='${test.path.name.toServiceMessageSafeString()}'")
+    }
+
+    override fun testExecutionFinish(test: TestScopeImpl, result: ExecutionResult) {
+        val name = test.path.name.toServiceMessageSafeString()
+        if (result is ExecutionResult.Failure) {
+            val exceptionDetails = getExceptionDetails(result)
+            out("testFailed name='$name' message='${exceptionDetails.first}' details='${exceptionDetails.second}'")
+
+        }
+        out("testFinished name='$name'")
+    }
+
+    override fun testIgnored(test: TestScopeImpl, reason: String?) {
+        val name = test.path.name.toServiceMessageSafeString()
+        out("testIgnored name='$name' ignoreComment='${reason ?: "no reason provided"}'")
+        out("testFinished name='$name'")
+    }
+
+    override fun groupExecutionStart(group: GroupScopeImpl) {
+        out("testSuiteStarted name='${group.path.name.toServiceMessageSafeString()}'")
+    }
+
+    override fun groupExecutionFinish(group: GroupScopeImpl, result: ExecutionResult) {

```

Comment:
Seems to be mostly the same as `groupExecutionFinish` (as other methods probably as well). This is one of reasons why I don’t like having `Action` instead of just `Group` — too much movements for a limited non-scalable usecase.

Reply:
This is a discussion for another time. :)